### PR TITLE
Adding JWT verifier.

### PR DESCRIPTION
**Pull request by jboeuf**
Still missing:
- Caching of the already checked JWTs (although it could be done at an upper layer).
- Caching of the jwks_uri to avoid 2 roundtrips for each verification.

Fixed:
- grpc_httpcli_context vs. grpc_pollset_set in credentials.c. @ctiller please check...

### CODE REVIEW

**Comment by jboeuf** [comment]
Maybe @nicolasnoble can have a look as well.

File: src/core/security/jwt_verifier.c [hunk 1]
```diff
@@ -0,0 +1,830 @@
...
+      req.path = gpr_strdup(GRPC_OPENID_CONFIG_URL_SUFFIX);
+    } else {
+      *(path_prefix++) = 0;
+      gpr_asprintf(&req.path, "/%s%s", path_prefix,
+                   GRPC_OPENID_CONFIG_URL_SUFFIX);
+    }
+    http_cb = on_openid_config_retrieved;
+  }
+
+  grpc_httpcli_get(&ctx->verifier->http_ctx, ctx->pollset, &req,
```
Comment (ctiller) [inline src/core/security/jwt_verifier.c]: lgtm

Comment (ctiller) [inline src/core/security/jwt_verifier.c]: (httpcli usage that is)

**Comment by ctiller** [comment]
Please hold off on merging this one until we get the ASAN problems sorted

**Comment by ctiller** [comment]
LGTM. Looks like it needs to be updated, but will merge when that's green.

**Comment by vjpai** [comment]
Fixes #2334

**Comment by jboeuf** [comment]
Jenkins is green.

**Pull request merged**